java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -1334,4 +1335,29 @@ public void copyFromSafe(int fromIndex, int thisIndex, BaseVariableWidthVector f
     }
     lastSet = thisIndex;
   }
+
+  @Override
+  public ArrowBufPointer getDataPointer(int index) {
+    ArrowBufPointer ret = new ArrowBufPointer();
```

Comment:
avoid duplication by 

getDataPointer(index, new ArrowBufPointer())

Reply:
Revised. Thank you so much.